packages/vsce/tests/SshConfigUtils.test.ts

```diff
@@ -0,0 +1,383 @@
+            (getVsceConfig as Mock).mockReturnValue({ get: vi.fn().mockReturnValue({}) });
+            process.env.ZOWE_OPT_SERVER_PATH = "/env/path";
+
+            const result = SshConfigUtils.getServerPath(profile);
+            expect(result).toBe("/env/path");
+        });
+        it("returns default path if nothing else is set", () => {
+            const profile: IProfile = { host: "testHost" } as IProfile;
+            (getVsceConfig as Mock).mockReturnValue({ get: vi.fn().mockReturnValue({}) });
+
+            const result = SshConfigUtils.getServerPath(profile);
+            expect(result).toBe(defaultPath);
+        });
+    });
+    describe("showSessionInTree", () => {
+        let mockApi: any;
+        let mockProvider: any;
+        let mockLocalStorage: any;
+        beforeEach(() => {
+            mockLocalStorage = {
+                getValue: vi.fn().mockReturnValue({ sessions: [] }),
+                setValue: vi.fn(),
+            };
+
+            mockProvider = {
+                mSessionNodes: [],
+                addSession: vi.fn(),
+                deleteSession: vi.fn(),
+                getTreeType: vi.fn().mockReturnValue("datasetProvider"),
+            };
+
+            mockApi = {
+                datasetProvider: mockProvider,
+                ussFileProvider: { ...mockProvider },
+                jobsProvider: { ...mockProvider },
+                getLocalStorage: vi.fn().mockReturnValue(mockLocalStorage),
+            };
+            vi.spyOn(ZoweVsCodeExtension, "getZoweExplorerApi").mockReturnValue({
+                getExplorerExtenderApi: () => mockApi,
+            } as any);
+        });
+        afterEach(() => {
+            vi.restoreAllMocks();
+        });
+        it("should add a session when visible is true and session is not present", async () => {
+            await SshConfigUtils.showSessionInTree("testProfile", true);
+
+            expect(mockProvider.addSession).toHaveBeenCalledWith(
+                expect.objectContaining({ sessionName: "testProfile", profileType: "ssh" }),
+            );
+            expect(mockLocalStorage.setValue).toHaveBeenCalledWith(
+                "datasetProvider",
+                expect.objectContaining({ sessions: ["testProfile"] }),
+            );
+        });
+        it("should delete a session when visible is false and session is present", async () => {
+            const fakeNode = { getProfileName: () => "testProfile" };
+            mockProvider.mSessionNodes = [fakeNode];
+
+            await SshConfigUtils.showSessionInTree("testProfile", false);
+
+            expect(mockProvider.deleteSession).toHaveBeenCalledWith(fakeNode);
+            expect(mockLocalStorage.setValue).toHaveBeenCalledWith(
+                "datasetProvider",
+                expect.objectContaining({ sessions: [] }),
+            );
+        });
+    });
+    describe("VscePromptApi", () => {
+        class TestVscePromptApi extends VscePromptApi {
```

Comment:
(nitpick) This is a creative approach to working around the constructor parameter for `VscePromptApi`. But this derived class is likely unnecessary if we just create an instance of `VscePromptApi` with the same parameter as what's passed into `super` 😋